lib/mobility-core/src/Kernel/Beam/Functions.hs

```diff
@@ -563,32 +587,327 @@ deleteInternal ::
+      throwError $ InternalError (("ART_DATA_PARSE_ERROR_OR_NOT_FOUND_CONDITIONAL_FIND " :: Text) <> show err)
+    Right artData' -> do
+      let artDataList = map (\artdata -> (ART.queryData artdata, ART.timestamp artdata, ART.requestId artdata)) artData'
+      queryData' <- getArtQueryObject "findAllWithKVAndConditionalDBInternal" tableName schemaName whereClause artDataList
+      case queryData' of
+        Nothing -> do
+          L.logError ("ART_QUERY_DATA_NOT_FOUND_CONDITIONAL_FIND" :: Text) $ "Art query data not found for table: " <> tableName <> " schema: " <> fromMaybe "" schemaName <> " where: " <> show whereClause
+          pure []
+        Just queryData'' -> do
+          let tableObject' = tableObject queryData''
+              schemaName' = ART.schemaName queryData''
+              tableName' = table queryData''
+          if tableName' == tableName && schemaName' == schemaName
+            then do
+              let tableIdentity = map decodeFromTextArt tableObject' :: [Maybe (table Identity)]
+              let tableIdentity' = catMaybes tableIdentity
+              case tableIdentity' of
+                [] -> do
+                  when (not $ null tableObject') $ L.logError ("ART_TABLE_OBJECT_NOT_FOUND_CONDITIONAL_FIND" :: Text) $ "Art table object not found for table: " <> tableName <> " schema: " <> fromMaybe "" schemaName <> " where: " <> show whereClause
+                  pure []
+                _ -> do
+                  now <- getCurrentTime
+                  void $ logQueryData "findAllWithKVAndConditionalDBInternalArt" where' [] tableIdentity' (meshEnabled meshConfig) tableName schemaName now
+                  logDebug $ "Found table identity for " <> tableName <> " schema: " <> fromMaybe "" schemaName <> " where: " <> show whereClause <> " tableIdentity: " <> show tableIdentity'
+                  res <- mapM ttype tableIdentity'
+                  pure $ catMaybes res
+            else do
+              L.logError ("ART_TABLE_MISMATCH_CONDITIONAL_FIND" :: Text) $ "Art table mismatch for table: " <> tableName <> " schema: " <> fromMaybe "" schemaName <> " where: " <> show whereClause
+              throwError $ InternalError ("Data found for different table : " <> tableName' <> " schema: " <> fromMaybe "" schemaName' <> " where: actual table and schema: " <> tableName <> " || " <> fromMaybe "" schemaName <> " where: " <> show whereClause)
+
+updateInternalArt ::
+  forall table m r.
+  (BeamTableFlow table m, EsqDBFlow m r) =>
+  MeshConfig ->
+  [Set Postgres table] ->
+  Where Postgres table ->
+  m ()
+updateInternalArt meshConfig' setClause whereClause = do
+  now <- getCurrentTime
+  void $ logQueryData "updateInternalArt" whereClause setClause [] (meshEnabled meshConfig') (modelTableName @table) (modelSchemaName @table) now
```

Comment:
why there is no other logic except logging for write queries?

Reply:
we are mocking the response of find queries by reading it from a file..while in case of write we just need to return the void as we don't require any data to work in the palace where it's being called.